contrib/epee/src/mlog.cpp

```diff
@@ -134,9 +135,35 @@ void mlog_configure(const std::string &filename_base, bool console, const std::s
   el::Loggers::addFlag(el::LoggingFlag::DisableApplicationAbortOnFatalLog);
   el::Loggers::addFlag(el::LoggingFlag::ColoredTerminalOutput);
   el::Loggers::addFlag(el::LoggingFlag::StrictLogFileSizeCheck);
-  el::Helpers::installPreRollOutCallback([filename_base](const char *name, size_t){
+  el::Helpers::installPreRollOutCallback([filename_base, max_log_rotation](const char *name, size_t){
     std::string rname = generate_log_filename(filename_base.c_str());
     rename(name, rname.c_str());
+    if (max_log_rotation != 0)
+    {
+      std::vector<boost::filesystem::path> found_files;
+      const boost::filesystem::directory_iterator end_itr;
+      for (boost::filesystem::directory_iterator iter(boost::filesystem::path(filename_base).parent_path()); iter != end_itr; ++iter)
+      {
+        const std::string filename = iter->path().string();
+        if (filename.size() > filename_base.size() && filename.substr(0, filename_base.size()) == filename_base)
+        {
+          found_files.push_back(iter->path());
+        }
+      }
+      if (found_files.size() > max_log_rotation)
+      {
+        std::sort(found_files.begin(), found_files.end(), [](const boost::filesystem::path &a, const boost::filesystem::path &b) {
+          time_t ta, tb;
+          file_io_utils::get_file_time(a.string(), ta);
+          file_io_utils::get_file_time(b.string(), tb);
+          return ta < tb;
+        });
+        for (size_t i = 0; i < found_files.size() - max_log_rotation; ++i)
```

Comment:
This is an unsigned comparision, so if found_files.size() is 10, i < some_very_large_number, it should end up dereferencing unallocated memory.
Maybe: for (size_t i = 0; found_files.size() - i > max_log_rotation; ++i) (also guards against huge max_log_rotation)

Reply:
There's already this check `if (found_files.size() > max_log_files)` just above, so I'm not sure why your concern applies. Or am I missing something?

Reply:
No, I had missed it.